### PR TITLE
dismiss on tap can be disabled by setting a preferences property

### PR DESCRIPTION
**Pull request by gurhankokcu**

### CODE REVIEW

File: Source/EasyTipView.swift [hunk 1]
```diff
@@ -408,7 +409,9 @@ open class EasyTipView: UIView {
     // MARK:- Callbacks -
     
     func handleTap() {
-        dismiss()
+        if preferences.dismissOnTap {
```
Comment (teodorpatras) [inline Source/EasyTipView.swift]: I would prefer it done differently:

On line 110, where the `UITapGestureRecognizer` gets created, I would set the tip view as it's delegate and then in the `gestureRecognizerShouldBegin(_:)` function, I would `return preferences.dismissOnTap`.

That's a cleaner approach.

File: Source/EasyTipView.swift [hunk 2]
```diff
@@ -200,6 +200,7 @@ open class EasyTipView: UIView {
         public var drawing      = Drawing()
         public var positioning  = Positioning()
         public var animating    = Animating()
+        public var dismissOnTap = true
```
Comment (teodorpatras) [inline Source/EasyTipView.swift]: please add this flag inside of the `Animating` struct.

File: Example/EasyTipView/ViewController.swift [hunk 3]
```diff
@@ -174,10 +174,11 @@ class ViewController: UIViewController, EasyTipViewDelegate {
             preferences.animating.showInitialAlpha = 0
             preferences.animating.showDuration = 1
             preferences.animating.dismissDuration = 1
+            preferences.animating.dismissOnTap = false
```
Comment (teodorpatras) [inline Example/EasyTipView/ViewController.swift]: the default value should be true, right?

**Comment by teodorpatras** [comment]
Thanks for your contribution!

**Comment by thepinyo** [comment]
what version for this merge?

**Comment by thepinyo** [comment]
now i'm can handle tap when set dismissOnTap to false. please change logic for handle this case.

**Pull request merged**